core/api/src/app/wallets/get-transactions-by-addresses.ts

```diff
@@ -1,61 +1,60 @@
-import { memoSharingConfig } from "@/config"
-import { PartialResult } from "@/app/partial-result"
+import { MAX_PAGINATION_PAGE_SIZE, memoSharingConfig } from "@/config"
 
 import { LedgerError } from "@/domain/ledger"
 import { WalletTransactionHistory } from "@/domain/wallets"
-import { CouldNotFindError } from "@/domain/errors"
 
 import { getNonEndUserWalletIds, LedgerService } from "@/services/ledger"
-import { WalletOnChainPendingReceiveRepository } from "@/services/mongoose"
+import { checkedToPaginatedQueryArgs } from "@/domain/primitives"
 
 export const getTransactionsForWalletsByAddresses = async ({
   wallets,
   addresses,
-  paginationArgs,
+  rawPaginationArgs,
 }: {
   wallets: Wallet[]
   addresses: OnChainAddress[]
-  paginationArgs?: PaginationArgs
-}): Promise<PartialResult<PaginatedArray<WalletTransaction>>> => {
-  const walletIds = wallets.map((wallet) => wallet.id)
+  rawPaginationArgs: {
+    first?: number
+    last?: number
+    before?: string
+    after?: string
+  }
```

Comment:
since it is used in different files probably is better to create a type

Reply:
Will make that change.